azurerm/internal/services/sentinel/sentinel_data_connector_office_365.go

```diff
@@ -0,0 +1,244 @@
+	defer cancel()
+
+	workspaceId, err := loganalyticsParse.LogAnalyticsWorkspaceID(d.Get("log_analytics_workspace_id").(string))
+	if err != nil {
+		return err
+	}
+	name := d.Get("name").(string)
+	id := parse.NewDataConnectorID(workspaceId.SubscriptionId, workspaceId.ResourceGroup, workspaceId.WorkspaceName, name)
+
+	if d.IsNewResource() {
+		resp, err := client.Get(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("checking for existing Sentinel Data Connector Office 365 %q: %+v", id, err)
+			}
+		}
+
+		id := dataConnectorID(resp.Value)
+		if id != nil && *id != "" {
+			return tf.ImportAsExistsError("azurerm_sentinel_data_connector_office_365", *id)
+		}
+	}
+
+	tenantId := d.Get("tenant_id").(string)
+	if tenantId == "" {
+		tenantId = meta.(*clients.Client).Account.TenantId
+	}
+
+	exchangeEnabled := d.Get("exchange_enabled").(bool)
+	sharePointEnabled := d.Get("share_point_enabled").(bool)
+	teamsEnabled := d.Get("teams_enabled").(bool)
+
+	// Service will not create the DC in case non of the toggle is enabled.
+	if !exchangeEnabled && !sharePointEnabled && !teamsEnabled {
+		return fmt.Errorf("one of `exchange_enabled`, `share_point_enabled` and `teams_enabled` should be `true`")
+	}
+
+	exchangeState := securityinsight.Enabled
+	if !exchangeEnabled {
+		exchangeState = securityinsight.Disabled
+	}
+
+	sharePointState := securityinsight.Enabled
+	if !sharePointEnabled {
+		sharePointState = securityinsight.Disabled
+	}
+
+	teamsState := securityinsight.Enabled
+	if !teamsEnabled {
+		teamsState = securityinsight.Disabled
+	}
+
+	param := securityinsight.OfficeDataConnector{
+		Name: &name,
+		OfficeDataConnectorProperties: &securityinsight.OfficeDataConnectorProperties{
+			TenantID: &tenantId,
+			DataTypes: &securityinsight.OfficeDataConnectorDataTypes{
+				Exchange: &securityinsight.OfficeDataConnectorDataTypesExchange{
+					State: exchangeState,
+				},
+				SharePoint: &securityinsight.OfficeDataConnectorDataTypesSharePoint{
+					State: sharePointState,
+				},
+				Teams: &securityinsight.OfficeDataConnectorDataTypesTeams{
+					State: teamsState,
+				},
+			},
+		},
+		Kind: securityinsight.KindOffice365,
+	}
+
+	// Service avoid concurrent updates of this resource via checking the "etag" to guarantee it is the same value as last Read.
+	if !d.IsNewResource() {
+		resp, err := client.Get(ctx, id.ResourceGroup, operationalInsightsResourceProvider, id.WorkspaceName, name)
+		if err != nil {
+			return fmt.Errorf("retrieving Sentinel Data Connector Office 365 %q: %+v", id, err)
+		}
+
+		if err := assertDataConnectorKind(resp.Value, securityinsight.DataConnectorKindOffice365); err != nil {
+			return fmt.Errorf("asserting Sentinel Data Connector of %q: %+v", id, err)
+		}
+		param.Etag = resp.Value.(securityinsight.OfficeDataConnector).Etag
```

Comment:
whilst there's good intentions here - the users confirmed that they want these changes deployed via `terraform plan` - so I don't think we need the Etag here?

Reply:
To set the etag based from the latest "read" before doing a "write" is mandatory for sentinel resources. I'm not quite get your meaning above?

Reply:
my understanding is the Etag is always optional in ARM - so that'd be worth opening an API bug for?

Reply:
Do you have some reference that stating the etag is optional in ARM? I explored the [azure api guide](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md) a bit and didn't find that. If the Etag is mandatory for updating sentinel resources, it might indicate that the service team always use the optimistic concurrency control?

Reply:
Based on the phrasing there

> optimistic concurrency control.

That matches my understanding that this is optimistic (e.g. optional) - whilst it's not called out specifically it does appear to match a conversation I recall having with the ARM Team

Reply:
This question hasn't been addressed yet?

Reply:
since this is being fixed on the service side - can we open an API bug tracking this?